runtime/metricsview/astsql.go

```diff
@@ -198,16 +231,26 @@ func (b *sqlBuilder) writeSelect(n *SelectNode) error {
 	return nil
 }
 
-func (b *sqlBuilder) writeJoin(joinType JoinType, baseSelect, joinSelect *SelectNode) error {
+func (b *sqlBuilder) writeJoin(joinType JoinType, baseSelect, joinSelect *SelectNode, joinTable *string) error {
```

Comment:
A little worried if this is becoming too hard to understand. I think the basic problem is that `SelectNode` is a fixed type, but that also keeps the code simple many other places, so changing it to an interface would have separate pains. 

What do you think about adding a `IsCTE bool` field to `SelectNode`? And then if it's set, then the node's alias is templated into the query instead of expanding the full `SELECT` subquery (everywhere except in the `WITH ...` part of course)?